lib/render/RayCaster.cpp

```diff
@@ -232,27 +267,31 @@ RayCaster::UserCoordinates::GetCurrentGrid( const RayCasterParams* params,
     {
         MyBase::SetErrMsg("UserCoordinates::GetCurrentGrid() isn't on a StructuredGrid; "
                           "the behavior is undefined in this case.");
+        return 1;
+    }
+    else
+    {
+        *gridpp = grid;
+        return 0;
     }
-    return grid;
 }
 
-bool RayCaster::UserCoordinates::IsUpToDate( const RayCasterParams* params,  
-                                                   DataMgr*         dataMgr ) const
+int  RayCaster::UserCoordinates::IsMetadataUpToDate( const RayCasterParams* params,  
```

Comment:
Since this function has multiple return codes, I would recommend using an enum instead of arbitrary integers

Reply:
I don't want to change the type of returned value from int to enum, so I instead used macros, see commit 03bed17f. Does this address your concern?

Reply:
Use asserts for progammatic errors (checking for min and max extent vector size())